Update package.md, add a test and a simple mock service (#2)
ballerina-platform/module-ballerinax-hubspot.crm.pipelines: README.md

## Overview

> [HubSpot](https://www.hubspot.com/our-story) is an AI-powered customer relationship management (CRM) platform. 

Reviewer: ```suggestion
[HubSpot](https://www.hubspot.com) is an AI-powered customer relationship management (CRM) platform. 
```

> The `ballerinax/hubspot.crm.pipelines` offers APIs to connect and interact with the [Hubspot Pipelines API](https://developers.hubspot.com/docs/guides/api/crm/pipelines) endpoints, specifically based on the [Hubspot Pipelines REST API](https://developers.hubspot.com/docs/reference/api/overview).

Reviewer: ```suggestion
The `ballerinax/hubspot.crm.pipelines` offers APIs to connect and interact with the [HubSpot Pipelines API](https://developers.hubspot.com/docs/guides/api/crm/pipelines) endpoints, specifically based on the [HubSpot REST API](https://developers.hubspot.com/docs/reference/api/overview).
```

   ```json
   {
     "token_type": "bearer",
     "refresh_token": "<Refresh Token>",
     "access_token": "<Access Token>",
     "expires_in": 1800
   }
   ```

5. Store the access token securely for use in your application.

## Quickstart

To use the HubSpot CRM Pipelines connector in your Ballerina application, follow these steps:

### Step 1: Import the module
```ballerina
import ballerinax/hubspot.crm.pipelines as hspipelines;
```

### Step 2: Instantiate a new connector

1. Create a `Config.toml` file and configure the obtained credentials:
```toml
   clientId = "<Client Id>"
   clientSecret = "<Client Secret>"
   refreshToken = "<Refresh Token>"
```
2. Create a `hspipelines:ConnectionConfig` with the obtained access token and initialize the connector with it.

### Step 3: Use Connector Operations

Now, utilize the available connector operations.

## Examples

The `HubSpot CRM Pipelines` connector provides practical examples illustrating usage in various scenarios. Explore these [examples](https://github.com/module-ballerinax-hubspot.crm.pipelines/tree/main/examples/), covering the following use cases:

1. [Pipeline management](examples/Pipeline-management)

> 2. [Support pipeline](examples/Support-pipeline)

Reviewer: Use the complete URL here.

Author: okay will do.

## Overview

[HubSpot](https://www.hubspot.com/our-story) is an AI-powered customer relationship management (CRM) platform. 

> The `ballerinax/hubspot.crm.pipelines` offers APIs to connect and interact with the [Hubspot Pipelines API](https://developers.hubspot.com/docs/guides/api/crm/pipelines)  endpoints, specifically based on the [OpenAPI specification for Hubspot Pipelines API v3](https://developers.hubspot.com/docs/reference/api/overview).

Reviewer: ```suggestion
The `ballerinax/hubspot.crm.pipelines` offers APIs to connect and interact with the [Hubspot Pipelines API](https://developers.hubspot.com/docs/guides/api/crm/pipelines) endpoints, specifically based on the [Hubspot Pipelines REST API](https://developers.hubspot.com/docs/reference/api/overview).
```

## Setup guide

To use the `HubSpot CRM Pipelines` connector, you must have access to the HubSpot API through a HubSpot developer account and a HubSpot App under it. Therefore you need to register for a developer account at HubSpot if you don't have one already.

### Step 1: Create/Login to a HubSpot Developer Account

If you have an account already, go to the [HubSpot developer portal](https://app.hubspot.com/)

If you don't have a HubSpot Developer Account you can sign up to a free account [here](https://developers.hubspot.com/get-started)

### Step 2 (Optional): Create a [Developer Test Account](https://developers.hubspot.com/beta-docs/getting-started/account-types#developer-test-accounts) under your account

Within app developer accounts, you can create developer test accounts to test apps and integrations without affecting any real HubSpot data.

> **_These accounts are only for development and testing purposes. In production, you should not use Developer Test Accounts._**

Reviewer: ```suggestion
### Step 2 (Optional): Create a Developer Test Account

Within app developer accounts, you can create a [developer test account](https://developers.hubspot.com/beta-docs/getting-started/account-types#developer-test-accounts) under your account to test apps and integrations without affecting any real HubSpot data.

> **Note:** These accounts are only for development and testing purposes. In production, you should not use Developer Test Accounts.
```

1. Go to the Test Account section from the left sidebar.

>    <img src="https://github.com/3nethz/module-ballerinax-hubspot.crm.pipelines/blob/main/docs/resources/test_acc_1.png" alt="Hubspot developer portal" style="width: 70%;">

Reviewer: Let's use the markdown format for images. Check the other places as well.

```suggestion
   ![Hubspot developer portal](docs/resources/test_acc_1.png)
```

   <img src="https://github.com/3nethz/module-ballerinax-hubspot.crm.pipelines/blob/main/docs/resources/create_app_final.png" alt="Hubspot app creation 1 testacc3" style="width: 70%;">

### Step 5: Get your Client ID and Client Secret

- Navigate to the Auth section of your app. Make sure to save the provided Client ID and Client Secret.

   <img src="https://github.com/3nethz/module-ballerinax-hubspot.crm.pipelines/blob/main/docs/resources/get_credentials.png" alt="Hubspot app creation 1 testacc3" style="width: 70%;">

### Step 6: Setup Authentication Flow

Before proceeding with the Quickstart, ensure you have obtained the Access Token using the following steps:

1. Create an authorization URL using the following format:

   ```
   https://app.hubspot.com/oauth/authorize?client_id=<YOUR_CLIENT_ID>&scope=<YOUR_SCOPES>&redirect_uri=<YOUR_REDIRECT_URI>
   ```

   Replace the `<YOUR_CLIENT_ID>`, `<YOUR_REDIRECT_URI>` and `<YOUR_SCOPES>` with your specific value.

2. Paste it in the browser and select your developer test account to intall the app when prompted.

3. A code will be displayed in the browser. Copy the code.

   ```
   Received code: na1-129d-860c-xxxx-xxxx-xxxxxxxxxxxx

>    ```

Reviewer: Remove this part.

```suggestion
```

   This command will return the access token necessary for API calls.

   ```json
   {
     "token_type": "bearer",
     "refresh_token": "<Refresh Token>",
     "access_token": "<Access Token>",
     "expires_in": 1800
   }
   ```

5. Store the access token securely for use in your application.

## Quickstart

To use the HubSpot CRM Pipelines connector in your Ballerina application, follow these steps:

### Step 1: Import the module
```ballerina
import ballerinax/hubspot.crm.pipelines as hspipelines;
```

### Step 2: Instantiate a new connector

1. Create a `Config.toml` file and configure the obtained credentials:
```toml
   clientId = "<Client Id>"
   clientSecret = "<Client Secret>"
   refreshToken = "<Refresh Token>"
```
2. Create a `hspipelines:ConnectionConfig` with the obtained access token and initialize the connector with it.

### Step 3: Use Connector Operations

Now, utilize the available connector operations.

## Examples

The `HubSpot CRM Pipelines` connector provides practical examples illustrating usage in various scenarios. Explore these [examples](https://github.com/module-ballerinax-hubspot.crm.pipelines/tree/main/examples/), covering the following use cases:

> 1. [Pipeline management](../examples/Pipeline-management/main.bal)
> 2. [Support pipeline](../examples/Support-pipeline/main.bal)

Reviewer: These paths seems to be incorrect. Please check whether they are working.
```suggestion
1. [Pipeline management](examples/Pipeline-management)
2. [Support pipeline](examples/Support-pipeline)
```
